pallets/author-noting/src/lib.rs

```diff
@@ -151,15 +156,39 @@ pub mod pallet {
                     relay_storage_proof,
                 } = data;
 
-                let relay_storage_root =
-                    T::RelayChainStateProvider::current_relay_chain_state().state_root;
+                let relay_chain_state = T::RelayChainStateProvider::current_relay_chain_state();
+                let relay_storage_root = relay_chain_state.state_root;
                 let relay_storage_rooted_proof =
                     GenericStateProof::new(relay_storage_root, relay_storage_proof)
                         .expect("Invalid relay chain state proof");
 
                 for para_id in registered_para_ids {
                     match Self::fetch_block_info_from_proof(&relay_storage_rooted_proof, para_id) {
-                        Ok(block_info) => LatestAuthor::<T>::insert(para_id, block_info),
+                        Ok(block_info) => {
+                            LatestAuthor::<T>::mutate(
+                                para_id,
+                                |maybe_old_block_info: &mut Option<ContainerChainBlockInfo<T>>| {
+                                    if let Some(ref mut old_block_info) = maybe_old_block_info {
+                                        if block_info.block_number > old_block_info.block_number {
+                                            // We only reward author if the block increases
+                                            total_weight = total_weight.saturating_add(
+                                                T::AuthorNotingHook::on_container_author_noted(
+                                                    &block_info.author,
+                                                    block_info.block_number,
+                                                    para_id,
+                                                ),
+                                            );
+                                            let _ = core::mem::replace(old_block_info, block_info);
+                                        }
+                                    } else {
+                                        let _ = core::mem::replace(
+                                            maybe_old_block_info,
```

Comment:
We don't reward author for the first block of this para_id?

Reply:
Indeed, we should differentiate between the case where it's really the 1st block in the container chain and the case where this container chain has been removed from the LatestAuthor storage item because it stopped producing blocks a long time ago (We don't have a LatestAuthor pruning strategy at the moment, but we'll probably have one int he future).
So, we should check that `block_info.block_number` is zero

Reply:
@nanocryk I finally decided to reward the author is there is no previous block, because we set the LatestAuthor at the same time so it can't be abuse for more than 1 block after pruning

Reply:
we should also put a TODO to inject a vector to `on_container_author_noted`, to better handle the weight estimations